e2e/jest.test-setup.ts

```diff
@@ -11,60 +8,87 @@ const isDebugMode = process.argv.includes('--debug');
-    const host = requestUrl.hostname;
-    // to improve page load performance, block network requests unrelated to application.
     try {
-      if (host === 'www.google-analytics.com'
-           || host === 'accounts.youtube.com'
-           || host === 'static.zdassets.com'
-           || host === 'play.google.com'
-           || request.url().endsWith('content-security-index-report')) {
-        request.abort();
-      } else {
-        request.continue();
+      request.continue();
+      // tslint:disable-next-line:no-empty
+    } catch (e) {
+    }
+  });
+
+  page.on('console', async(message) => {
+    if (message != null) {
+      const msg = message.type().toUpperCase();
+      // Don't log "log", "info" or "debug"
+      if (msg.includes('ERROR') || msg.includes('WARNING')) {
+        console.debug(`❗ ${message.type()}: ${message.text()}`);
+      }
+    }
+  });
+
+  // Emitted when the page crashed
+  page.on('error', error => {
+    console.debug(`❗ ${error}`);
+  });
+
+  // Emitted when a script has uncaught exception
+  page.on('pageerror', error => {
+    if (error != null) {
+      console.debug(`❗ ${error}`);
+    }
+  });
+
+  // Emitted when a request failed. Warning: blocked requests from above will be logged as failed requests, safe to ignore these.
+  page.on('requestfailed', async request => {
+    try {
+      const response = request.response();
+      if (response !== null) {
+        const status = response.status();
+        const responseText = await response.text();
+        const failureError = request.failure().errorText;
+        console.debug(`❗ ${status} ${request.method()} ${request.url()}  \n ${failureError} \n ${responseText}`);
       }
+      // tslint:disable-next-line:no-empty
     } catch (err) {
-      console.error(err);
     }
   });
-  if (isDebugMode) {
-    // Emitted when a request failed. Warning: blocked requests from above will be logged as failed requests, safe to ignore these.
-    page.on('requestfailed', request => {
-      console.error(`❌ Failed request => ${request.method()} ${request.url()}`);
-      request.continue();
-    });
-    // Emitted when the page crashed
-    page.on('error', error => console.error(`❌ ${error}`));
-    // Emitted when a script has uncaught exception
-    page.on('pageerror', error => console.error(`❌ ${error}`));
-  }
+
+  page.on('response', async(response) => {
+    try {
+      const request = response.request();
+      const requestUrl = request.url();
+
+      // Long only responses from AoU-app requests
+      if (requestUrl.includes('api-dot-all-of-us')) {
+        const failure = request.failure();
+        const method = request.method().trim();
+        if (method !== 'OPTIONS') {
+          if (failure !== null) {
+            console.debug(`❗ ${response.status()} ${method} ${requestUrl} \n ${failure.errorText}`);
+          } else {
+            console.debug(`❗ ${response.status()} ${request.method()} ${requestUrl}`);
```

Comment:
nit: this line prints `request.method()` while the one above prints just `method`. I know the same information is being conveyed but I feel like it's worth being consistent.